File: output/gpt_4o/avatar/direct_translation/java/python/codeforces_282_B.py

n = int(input())
ch = [''] * n
s1 = 0
s2 = 0
flag = 0

for i in range(n):
    x, y = map(int, input().split())
    temp1 = s1 + x
    temp2 = s2 + y
    if abs(temp1 - s2) <= 500:
        s1 += x
        ch[i] = 'A'
        continue
    if abs(temp2 - s1) <= 500:
        s2 += y
        ch[i] = 'G'
        continue
    flag = 1
    break

if flag == 1:
    print(-1)
else:
    ans = ''.join(ch)
    print(ans)

File: output/gpt_4o/avatar/direct_translation/python/java/codeforces_282_B.java

import java.util.Scanner;
import java.util.ArrayList;

public class codeforces_282_B {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int firstline = Integer.parseInt(scanner.nextLine());
        int total = 0;
        ArrayList<String> memory = new ArrayList<>();
        for (int x = 0; x < firstline; x++) {
            String[] input = scanner.nextLine().split(" ");
            int A = Integer.parseInt(input[0]);
            int G = Integer.parseInt(input[1]);
            if (total + A <= 500) {
                total += A;
                memory.add("A");
            } else {
                total -= G;
                memory.add("G");
            }
        }
        scanner.close();
        System.out.println(String.join("", memory));
    }
}

